src/moonlink/src/storage/mooncake_table/tests.rs

```diff
@@ -1142,38 +1219,1290 @@ async fn test_streaming_begin_flush_abort_end_flush_multiple() {
+
+    sync_mooncake_snapshot(&mut table, &mut event_completion_rx).await;
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_puffin_deletion_blob_inconsistency_assertion() -> Result<()> {
+    let context = TestContext::new("puffin_deletion_blob_inconsistency");
+    let mut table = test_table(
+        &context,
+        "puffin_deletion_blob_inconsistency",
+        IdentityProp::Keys(vec![0]),
+    )
+    .await;
+    let (event_completion_tx, mut event_completion_rx) = mpsc::channel(100);
+    table.register_table_notify(event_completion_tx).await;
+
+    // Step 1: Create initial data and flush it to create persistent data files
+    append_rows(
+        &mut table,
+        vec![
+            test_row(1, "Data1", 20),
+            test_row(2, "Data2", 21),
+            test_row(3, "Data3", 22),
+            test_row(4, "Data4", 23),
+        ],
+    )?;
+    table.commit(1);
+    flush_table_and_sync(&mut table, &mut event_completion_rx, 10).await?;
+
+    // Step 2: Delete some rows to create deletion vectors
+    table.delete(test_row(2, "Data2", 21), 11).await;
+    table.delete(test_row(3, "Data3", 22), 11).await;
+    table.commit(12);
+
+    // Step 3: Create and persist snapshot to create puffin deletion blobs
+    // This should persist the deletion vectors as puffin blobs
+    create_mooncake_and_persist_for_test(&mut table, &mut event_completion_rx).await;
+
+    // Step 4: Add more data to trigger compaction threshold
+    append_rows(
+        &mut table,
+        vec![
+            test_row(5, "Data5", 24),
+            test_row(6, "Data6", 25),
+            test_row(7, "Data7", 26),
+            test_row(8, "Data8", 27),
+        ],
+    )?;
+    table.commit(15);
+    flush_table_and_sync(&mut table, &mut event_completion_rx, 20).await?;
+
+    // Step 5: Create more deletions and persist again
+    table.delete(test_row(5, "Data5", 24), 21).await;
+    table.delete(test_row(6, "Data6", 25), 21).await;
+    table.commit(22);
+    create_mooncake_and_persist_for_test(&mut table, &mut event_completion_rx).await;
+
+    // Step 6: Add even more data to ensure we have enough files for compaction
+    append_rows(
+        &mut table,
+        vec![test_row(9, "Data9", 28), test_row(10, "Data10", 29)],
+    )?;
+    table.commit(27);
+    flush_table_and_sync(&mut table, &mut event_completion_rx, 26).await?;
+
+    // Step 7: Force data compaction
+    let created = table.create_snapshot(SnapshotOption {
+        uuid: uuid::Uuid::new_v4(),
+        force_create: true,
+        skip_iceberg_snapshot: true,
+        index_merge_option: MaintenanceOption::Skip,
+        data_compaction_option: MaintenanceOption::ForceRegular, // Force data compaction
+    });
+
+    assert!(created, "Mooncake snapshot should be created");
+
+    sync_mooncake_snapshot(&mut table, &mut event_completion_rx).await;
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_stream_commit_with_pending_flush_deletion_remapping_bug() -> Result<()> {
```

Comment:
why name the test case as `_bug`?

Reply:
If you think it's a bug fix, you need to link the issue around (duckdb does that)